### PR TITLE
Refactoring QueryStringSearchOperatorDsl

### PR DESCRIPTION
**Pull request by masonJS**
Hi, Jake Son.
I've made a few edits to the work I've done.
Please check and approve.

### CODE REVIEW

File: core/src/main/kotlin/com/github/inflab/spring/data/mongodb/core/aggregation/search/QueryStringQueryOptionDsl.kt [hunk 1]
```diff
@@ -41,7 +43,23 @@ class QueryStringQueryOptionDsl {
     fun text(value: String, field: String? = null): Query {
         val escaped = value.replace("*", "\\*").replace("?", "\\?")
 
-        return Query("\"$escaped\"", field)
+        query = Query("\"$escaped\"", field)
+
+        return query
```
Comment (jbl428) [inline core/src/main/kotlin/com/github/inflab/spring/data/mongodb/core/aggregation/search/QueryStringQueryOptionDsl.kt:48]: I think this idea is great. 
but there is another solution.

we can change `QueryStringSearchOperatorDsl.kt` like this.

```kt
    fun query(configuration: QueryStringQueryOptionDsl.() -> QueryStringQueryOptionDsl.Query) {
        val query = QueryStringQueryOptionDsl().configuration()
        document["query"] = query.toString()
    }
```

We no longer need to reassign the query in every function by using this method.
what do you think?

File: core/src/main/kotlin/com/github/inflab/spring/data/mongodb/core/aggregation/search/QueryStringQueryOptionDsl.kt [hunk 2]
```diff
@@ -88,7 +134,38 @@ class QueryStringQueryOptionDsl {
...
+    /**
+     * Creates a range query.
+     *
+     * @param left The left value to search
+     * @param right The right value to search
+     * @param leftInclusion The left value is included in the range
+     * @param rightInclusion The right value is included in the range
+     * @param field Indexed field to search
+     */
+    fun range(left: String, right: String, leftInclusion: Boolean = true, rightInclusion: Boolean = true, field: KProperty<String?>): Query {
```
Comment (jbl428) [inline core/src/main/kotlin/com/github/inflab/spring/data/mongodb/core/aggregation/search/QueryStringQueryOptionDsl.kt:151]: 👍 👍 👍 👍 👍

File: core/src/main/kotlin/com/github/inflab/spring/data/mongodb/core/aggregation/search/QueryStringQueryOptionDsl.kt (hunk 1 above)
Comment (haruInflearn) [reply core/src/main/kotlin/com/github/inflab/spring/data/mongodb/core/aggregation/search/QueryStringQueryOptionDsl.kt:48]: Thanks for the suggestion.
I'm going to ask you again just to make sure I understand what you mean.

If I write it as you say, 
the `query`property that existed in the `QueryStringQueryOptionDsl`class will disappear?

Comment (jbl428) [reply core/src/main/kotlin/com/github/inflab/spring/data/mongodb/core/aggregation/search/QueryStringQueryOptionDsl.kt:48]: Yes, and build method also removed.

**Pull request closed without merging**